src/parser.cpp

```diff
@@ -1010,7 +1010,47 @@ Writer<std::unique_ptr<AST::AST>> Parser::parse_type_term() {
 	return make_writer<std::unique_ptr<AST::AST>>(std::move(e));
 }
 
-Writer<std::unique_ptr<AST::AST>> Parser::parse_type_function() {
+Writer<std::unique_ptr<AST::AST>> Parser::parse_type_list(bool with_identifiers = false) {
```

Comment:
I know C++ is not very ergonomic for this kinda thing, but returning the two vectors (in a `std::pair`) would be nicer than creating a superfluous node

Reply:
```c++
Writer<std::pair<std::vector<std::unique_ptr<AST::AST>>, std::vector<std::unique_ptr<AST::AST>>>> Parser::parse_type_function() {}
```

Reply:
I agree, this is exactly what I needed. But I just ended up creating another node :(